staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt-config.go

```diff
@@ -0,0 +1,246 @@
+	reqMgmt.triggerReload()
+}
+
+func (reqMgmt *requestManagementSystem) Run(stopCh <-chan struct{}) error {
+	defer reqMgmt.configQueue.ShutDown()
+	klog.Info("Starting reqmgmt config controller")
+	if ok := cache.WaitForCacheSync(stopCh, reqMgmt.plInformerSynced, reqMgmt.fsInformerSynced); !ok {
+		return fmt.Errorf("Never achieved initial sync")
+	}
+	klog.Info("Running reqmgmt config worker")
+	wait.Until(reqMgmt.runWorker, time.Second, stopCh)
+	klog.Info("Shutting down reqmgmt config worker")
+	return nil
+}
+
+func (reqMgmt *requestManagementSystem) runWorker() {
+	for reqMgmt.processNextWorkItem() {
+	}
+}
+
+func (reqMgmt *requestManagementSystem) processNextWorkItem() bool {
+	obj, shutdown := reqMgmt.configQueue.Get()
+	if shutdown {
+		return false
+	}
+
+	func(obj interface{}) {
+		defer reqMgmt.configQueue.Done(obj)
+		if !reqMgmt.syncOne() {
+			reqMgmt.configQueue.AddRateLimited(obj)
+		} else {
+			reqMgmt.configQueue.Forget(obj)
+		}
+	}(obj)
+
+	return true
+}
+
+// syncOne attempts to sync all the config for the reqmgmt filter.  It
+// either succeeds and returns `true` or logs an error and returns
+// `false`.
+func (reqMgmt *requestManagementSystem) syncOne() bool {
+	all := labels.Everything()
+	newPLs, err := reqMgmt.plLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list PriorityLevelConfiguration objects: %s", err.Error())
+		return false
+	}
+	newFSs, err := reqMgmt.fsLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list FlowSchema objects: %s", err.Error())
+		return false
+	}
+	reqMgmt.digestConfigObjects(newPLs, newFSs)
+	return true
+}
+
+func (reqMgmt *requestManagementSystem) digestConfigObjects(newPLs []*rmtypesv1a1.PriorityLevelConfiguration, newFSs []*rmtypesv1a1.FlowSchema) {
+	oldRMState := reqMgmt.curState.Load().(*requestManagementState)
+	var shareSum float64
+	newRMState := &requestManagementState{
+		priorityLevelStates: make(map[string]*priorityLevelState),
+	}
+	newlyQuiescent := make([]*priorityLevelState, 0)
+	for _, pl := range newPLs {
+		state := oldRMState.priorityLevelStates[pl.Name]
+		if state == nil {
+			state = &priorityLevelState{
+				config: pl.Spec,
+			}
+			newRMState.priorityLevelStates[pl.Name] = state
+			continue
+		}
+
+		if state.config.Exempt {
+			continue
+		}
+
+		oState := *state
+		state = &oState
+		state.config = pl.Spec
+		if state.emptyHandler != nil { // it was undesired, but no longer
+			klog.V(3).Infof("Priority level %s was undesired and has become desired again", pl.Name)
+			state.emptyHandler = nil
+			state.queues.Quiesce(nil)
+		}
+		shareSum += float64(state.config.AssuredConcurrencyShares)
+		newRMState.priorityLevelStates[pl.Name] = state
+	}
+	fsSeq := make(FlowSchemaSequence, 0, len(newFSs))
+	for _, fs := range newFSs {
+		if !warnFlowSchemaSpec(fs.Name, &fs.Spec, newRMState.priorityLevelStates, oldRMState.priorityLevelStates) {
+			continue
+		}
+		fsSeq = append(fsSeq, fs)
+	}
+	sort.Sort(fsSeq)
+	if klog.V(8) {
```

Comment:
I think this should be more prominent than `V(8)`.  It is much rarer than every request, and has much greater impact than processing a single request.

Reply:
s/8/5/, wdyt?